File: Week6/sentimental-mario-less/mario.py
from cs50 import get_int


def printPyramid(h):

    for i in range(h):  # loop of rows
        
        n = h - i - 1
        
        for j in range(n):
            
            print(" ", end='')  # print spaces in a row

        for j in range(i+1):  # loop of # in a row

            print("#", end='')  # print # in a row

        print("")  # go next row
        

def main():

    minHeight = 1
    maxHeight = 8

    height = get_int("Type pyramid height:")

    while height < minHeight or height > maxHeight:  # reject wrong inputs and asks again.

        height = get_int("Type pyramid height:")

    printPyramid(height)


if __name__ == "__main__":
    main()

File: Week1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void printPyramid(int h);

int main(void)
{
    int height;
    int minHeight = 1;
    int maxHeight = 8;
    
    do
    {
        height = get_int("Type pyramid height:\n");
    }
    while (height < minHeight || height > maxHeight); // rejeita entradas indevidas e solicita novamente.
    
    printPyramid(height);
    
}

void printPyramid(int h)
{
    for (int i = 0; i < h; i++) //loop of rows
    {
        for (int j = 0; j < h - i - 1; j++) //loop of spaces in a row
        {
            printf(" "); //print spaces in a row
        }
        
        for (int j = 0; j <= i; j++) //loop of # in a row
        {
            printf("#"); //print # in a row
        }
        
        printf("\n"); //go next row
    }   

}
